File: Source/server/rabbit-game/src/SharedModel/ModelEvents/ClientIntention.cs
namespace RabbitGameServer.SharedModel.ModelEvents
{

	public enum ClientIntentionType
	{
		Move,
		Attack,
		Defend,
		ReadyForInit,
		AbortAttack
	}

	public class ClientIntention
	{

		public ClientIntentionType type { get; set; }
		public string playerName { get; set; }

		public ClientIntention(ClientIntentionType type, string playerName)
		{
			this.type = type;
			this.playerName = playerName;
		}
	}
}

File: Source/root/src/main/java/root/model/event/ClientIntention.java
package root.model.event;

public abstract class ClientIntention {

	public static enum ClientIntentionType {
		Move,
		Attack,
		Defend,
		ReadyForInit,
		AbortAttack
	}

	private ClientIntentionType type;

	private String playerName;

	public ClientIntention(ClientIntentionType type, String playerName) {
		this.type = type;
		this.setPlayerName(playerName);
	}

	public ClientIntentionType getEventType() {
		return type;
	}

	public void setEventType(ClientIntentionType type) {
		this.type = type;
	}

	public String getPlayerName() {
		return playerName;
	}

	public void setPlayerName(String playerName) {
		this.playerName = playerName;
	}

}
